CDCgov/prime-reportstream
# Update condition filter code to pass routing if message contains no observations

This PR ensure ETOR data can make it through all the way past the send step. Condition filter was the biggest blocker here. Since an OML may not contain any observations, we had to make an update to the routing condition filter to pass if no observations are found. This change logically makes sense.

Test Steps:
1. Follow instructions in #8676 to set up ETOR sender/receiver
2. Send the FHIR data below OR, better yet, the data in this PR: https://github.com/CDCgov/trusted-intermediary/pull/246
3. verify it makes it to the local SFTP bucket under build -> sftp

```
{"resourceType":"Bundle","identifier":{"value":"CDC_TI_OML.1.1234567"},"id":"df19c110-f28e-4277-9bec-95a9a7ec091e","type":"message","timestamp":"2023-02-24T09:55:17.713-06:00","entry":[{"resource":{"resourceType":"MessageHeader","id":"a53d27ee-a78c-4395-99b5-5fd66e8b96ba","meta":{"tag":[{"system":"http://terminology.hl7.org/CodeSystem/v2-0103","code":"P","display":"Production"}]},"eventCoding":{"system":"http://terminology.hl7.org/CodeSystem/v2-0003","code":"O21","display":"OML-Laboratoryorder"},"source":{"name":"CDCTrustedIntermediary","endpoint":"https://reportstream.cdc.gov/"}}},{"fullUrl":"Provenance/cd2e9813-d9cd-4cb9-bf05-7dcdb6b5fe10","resource":{"resourceType":"Provenance","id":"cd2e9813-d9cd-4cb9-bf05-7dcdb6b5fe10","recorded":"2023-03-02T21:30:42.888+00:00","activity":{"coding":[{"system":"http://terminology.hl7.org/CodeSystem/v2-0003","code":"O21","display":"OML-Laboratoryorder"}]}}},{"resource":{"resourceType":"Patient","id":"infant-twin-1","extension":[{"url":"http://hl7.org/fhir/us/core/StructureDefinition/us-core-race","extension":[{"url":"text","valueString":"Asian"}]}],"identifier":[{"type":{"coding":[{"system":"http://terminology.hl7.org/CodeSystem/v2-0203","code":"MR","display":"MedicalRecordNumber"}]},"value":"MRN7465737865"}],"name":[{"use":"official","family":"Solo","given":["Jaina"]}],"gender":"female","birthDate":"2017-05-15","_birthDate":{"extension":[{"url":"http://hl7.org/fhir/StructureDefinition/patient-birthTime","valueDateTime":"2017-05-15T11:11:00-05:00"}]},"multipleBirthInteger":1,"contact":[{"relationship":[{"coding":[{"system":"http://terminology.hl7.org/CodeSystem/v2-0131","code":"N","display":"Nextofkin"}]}],"name":{"family":"Organa","given":["Leia"]},"telecom":[{"system":"phone","value":"+31201234567"}]}]}},{"resource":{"resourceType":"ServiceRequest","id":"8ac2b99f-8d18-48f8-a798-bb0fb2d16e31","status":"active","intent":"order","code":{"coding":[{"system":"http://loinc.org","code":"54089-8","display":"NewbornScreeningPanel"}]},"subject":{"reference":"Patient/infant-twin-1"},"occurrenceDateTime":"2023-02-24T09:55:17-06:00"}}]}
```

## Changes
- Improved condition filter routing logic
- Added test for new condition filter routing logic
- Updated ETOR simulated lab receiver config

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #8676 

## To Be Done
- https://github.com/CDCgov/trusted-intermediary/pull/246
